lib/acfs/global.rb

```diff
@@ -83,7 +83,7 @@ def add_callback(resource, &block)
     def on(*resources)
       resources.each do |resource|
         add_callback resource do |_|
-          yield(*resources) unless resources.any? {|res| !res.loaded? }
+          yield(*resources) if resources.all? {|res| res.nil? || res.loaded? }
```

Comment:
Do we need the `nil?` check in `add_callback` as well? (You proposed that on your machine, IIRC.) I don't quite see the use-case. And it's not needed to satisfy the tests.

Reply:
It does iterate over `resources` too and calls `add_callback` for each. It should fail too.

Reply:
This only is an issue when the resource delegator from `#find_by` is already loaded but no resource find. The delegation target is replaced with `nil` then.

Reply:
Fair point and quite complicated, as the tests showed. :wink:

Reply:
Done.